tests/test_devshell.py

```diff
@@ -195,19 +201,25 @@ def test_no_refresh_token(self):
             creds = DevshellCredentials()
             self.assertEquals(None, creds.refresh_token)
 
-    def test_reads_credentials(self):
+    @mock.patch.object(devshell, '_UTCNOW')
+    def test_reads_credentials(self, utcnow):
+        NOW = datetime.datetime(1992, 12, 31)
+        utcnow.return_value = NOW
         with _AuthReferenceServer():
             creds = DevshellCredentials()
             self.assertEqual('joe@example.com', creds.user_email)
             self.assertEqual('fooproj', creds.project_id)
             self.assertEqual('sometoken', creds.access_token)
-
+            self.assertEqual(
+                NOW + datetime.timedelta(seconds=1000), creds.token_expiry)
+            
```

Comment:
You should check the calls to your stub. Something like

```python
self.assertEqual(utcnow.mock_calls, [...])
```

or `utcnow.assert_called_with` or something similar. (Check out the [docs][1] for other options.)

[1]: https://docs.python.org/3/library/unittest.mock.html